### DISCUSSION TITLE
cookies won't be automatically deleted when the "reject" button is pressed

### DISCUSSION THREAD

**Discussion by yuntaoii** [orestbida/cookieconsent#499] [Q&A]
https://stackblitz.com/edit/nextjs-euxk9k?file=components%2FCookieConsentComponent.js

I just copied the code from in this nextjs project, and i did few modifications in onChange and onAccept function as follows:
```
onAccept: function (cookie) {
    console.log("onAccept", cookie);
    const analyticsEnabled =
      window.CookieConsentApi.allowedCategory("analytics");
    if (analyticsEnabled) {
      console.log("Accepted!");

      gtag("consent", "update", {
        ad_storage: "granted",
        analytics_storage: "granted",
      });
    } else {
      console.log("Rejected!");
    }
  },

  onChange: function (cookie, changed_preferences) {
    console.log("onChange", cookie);
    const analyticsEnabled =
      window.CookieConsentApi.allowedCategory("analytics");
    if (analyticsEnabled) {
      console.log("Accepted!");

      gtag("consent", "update", {
        ad_storage: "granted",
        analytics_storage: "granted",
      });
    } else {
      console.log("Rejected!");
      gtag("consent", "update", {
        ad_storage: "denied",
        analytics_storage: "denied",
      });
    }
  },
```

The thing is, in the beginning, i pressed the "Accept all" button, and then i can see those cookies named "_ga" from Chrome Developer Tools. But then if i press the "Reject all" button from the settings modal, those cookies won't be deleted as i can still see them from Developer Tools, though i had thought that autoclear function would be called to delete them immediately. Only after i reload the page, those cookies will be cleared. is there something wrong?

**Comment by orestbida** [comment]
Hmm, your config. throws errors on my end. The [`demo_gtm example`](https://github.com/orestbida/cookieconsent/tree/master/demo/demo_gtm) works fine though.

Note that you can't enable/disable multiple times gtag (afaik) without a page reload.
